File: Week 1/Problem Set 1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int n;
    const int gap_size = 2;
    //Get height of pyramid from user
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    //making pyramid
    for (int i = 0; i < n; i++)
    {
        //Print spaces
        for (int j = 0; j < n - i - 1; j++)
        {
            printf(" ");
        }

        //print left hashes
        for (int j = n - i; j <= n; j++)
        {
            printf("#");
        }

        //print gap
        for (int j = 0; j < gap_size; j++)
        {
            printf(" ");
        }

        //print right hashes
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: Week 6/sentimental-mario-more/mario.py
def get_height():
    while True:
        try:
            n = int(input("Height: "))
            if n >= 1 and n <= 8:
                return n
        except ValueError:
            print("Not an integer")


def main():
    gap_size = 2
    height = get_height()
    for i in range(height):
        # print spaces
        for j in range(height - i - 1):
            print(" ", end="")

        # print left hashes
        for j in range(height - i, height + 1):
            print("#", end="")

        # print gap
        for j in range(gap_size):
            print(" ", end="")

        # print right hashes
        for j in range(i + 1):
            print("#", end="")

        print()


main()
